store/models/service_agreement.go

```diff
@@ -14,7 +14,14 @@ type ServiceAgreement struct {
 	Encumbrance Encumbrance `json:"encumbrance" storm:"inline"`
 	ID          string      `json:"id" storm:"id,unique"`
 	JobSpecID   string      `json:"jobSpecID"`
-	jobSpec     JobSpec
+	Normalized  string      `json:"normalizedRequest"`
```

Comment:
This looks great, but I don't fully grasp why we need the `Normalized` field on a service agreement. I assume it's to normalize the case sensitive key collision, which is a response concern, but that's not immediately obvious to another programmer. Could you elaborate on this? Perhaps find a way to have it self documented in code by introducing explanatory method names or a new response type. An idea: could this be handled exclusively in the `presenter` wrapper?

Reply:
Short answer: it's an extra precaution so that we can always have the interpretation of the request at the time it was sent in, which is part of what's covered in the signed commitment.

The normalized version of the original request is used to generate the digest, which in turn is used to determine the Service Agreement ID. It's possible to regenerate the normalized job spec from how we serialize it in the database, but that's a bit complicated. And, if we ever need to change our normalizing algorithm then all of the digests won't match anymore, so we'd have to keep historic versions. We could just store the digest, but it's possible we'll need the full agreement in the future, and it's nice to always be able to show the original agreement.

Another thing to note is that there are some parameters in the original request that could be dropped(anything that doesn't have a field in the DB). Holding onto those isn't crucial, but it's nice because they essentially act as nonces, or ways of introducing variations in IDs.